### PR TITLE
LRQA-73703 Add test AccountMemberCanSearchForAnAccount | master

### PR DESCRIPTION
**Pull request by epark18**
https://issues.liferay.com/browse/LRQA-73703

### CODE REVIEW

**Comment by epark18** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
Just starting review.

File: modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase
Comment (Sharryxue) [inline modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase]: - Could you please double check if the steps here can be replaced by the exist json macro? Like add account via json, and add account user via json. 
- If so, we probably also no need to login the admin user. 
- Since we are testing for the _account member_ permission, we can add a new user, and assign him at least 2 accounts for testing.
Thanks.

**Review by Sharryxue — CHANGES_REQUESTED** [review]
Hello @epark18,
Could you please double check the test when you get a chance? Thanks in advance.

File: modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase
Comment (epark18) [reply modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase]: I think there will be an issue assigning the user to different accounts through JSON. Right now, we have _JSONAccountEntryUser.addAccountEntryUser_ which creates a user AND adds that user into an account.
As far as I know, this macro can only be used to add multiple different users into any account, but not one user into multiple different accounts. 

I think I can try:
1. (Staying logged in as userea): Create a new account user for Account Name through JSON.
2. Create another account through JSON.
3. Assign new user to 2nd account through UI (2B page).

**Comment by epark18** [comment]
ci:test:sf

File: modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase
Comment (epark18) [reply modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase]: I have tried testing many times here: https://github.com/epark18/liferay-portal/pull/29 using the JSON macros while staying logged in as userea with no luck 😢 Main issue include assigning the new user to a 2nd account. Pushed up a working test, but I had to sign in as Test.

**Comment by epark18** [comment]
ci:test:account

**Comment by Sharryxue** [comment]
Just starting review.

File: modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase
Comment (Sharryxue) [inline modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase]: Move ahead into the Given task block, thanks.
Should the _property custom.properties = "jsonws.web.service.paths.excludes=";_ required for this macro?

File: modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase
Comment (Sharryxue) [inline modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase]: This is duplicate to next step.

File: modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase
Comment (Sharryxue) [inline modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase:784]: Would it be better to add a new account with no user, and confirm the account not shows up when the use login? Thanks.

**Review by Sharryxue — CHANGES_REQUESTED** [review]
Hello @epark18 ,
Sorry for a late review, please update the test to add a new account with no user to compare. Thanks so much.

**Comment by epark18** [comment]
Taking a look! Thank you.

File: modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase
Comment (epark18) [reply modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase:784]: I think that's covered in 2BAccount#CanAccessAListOfAccountsTheUserBelongsTo. For this test, I think just asserting the search bar works for accounts might be enough? I am pushing up the changes for your other comments, but will wait for your final decision on this. Thank you!

**Comment by epark18** [comment]
ci:test:sf

File: modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase
Comment (Sharryxue) [reply modules/apps/account/account-test/src/testFunctional/tests/2BAccount.testcase:784]: Cool, thanks for double checking it Eric, you're right, that would be duplicate to the other test.

**Comment by Sharryxue** [comment]
Just starting review.

**Comment by Sharryxue** [comment]
No changes needed. : )

**Comment by Sharryxue** [comment]
Rebase and resend.
https://github.com/Sharryxue/liferay-portal/pull/464

**Pull request closed without merging**
